ee/ucp/kubernetes/layer-7-routing.md

```diff
@@ -19,87 +21,180 @@ Use an ingress controller when you want to:
 * Give your Kubernetes app an externally-reachable URL.
 * Load-balance traffic to your app.
 
-Kubernetes provides an NGINX ingress controller that you can use in Docker EE
-without modifications.
-Learn about [ingress in Kubernetes](https://v1-8.docs.kubernetes.io/docs/concepts/services-networking/ingress/).
+Ingress is an API object that manages external access to the services in a cluster, typically HTTP. In UCP 3.1, Ingress must be deployed separately. There are a variety of ingress controllers, but for UCP, the supported one is the [NGINX Ingress Controller](https://github.com/kubernetes/ingress-nginx). The following instructions provide highly-available production deployment of this controller. 
 
-## Create a dedicated namespace
+## Prerequisites
 
-1.  Navigate to the **Namespaces** page, and click **Create**.
-2.  In the **Object YAML** editor, append the following text.
-    ```yaml
-    metadata:
-      name: ingress-nginx
-    ```
+- UCP 3.1.X deployed and properly configured
+- Two or three dedicated infrae nodes deployed as UCP worker nodes
+- An external load-balancer fronting these nodes with an associated VIP that resolves the application DNS (for example, `*.app.docker.mycompany.com`)
 
-    The finished YAML should look like this.
+### Step 1: Labeling infrastructure nodes
 
-    ```yaml
-    apiVersion: v1
-    kind: Namespace
-    metadata:
-      name: ingress-nginx
-    ```
-3.  Click **Create**.
-4.  In the **ingress-nginx** namespace, click the **More options** icon,
-    and in the context menu, select **Set Context**.
+To deploy the NGINX controller on the infra nodes, the nodes must be labeled. Identify 2 or 3 nodes as the infra nodes to host the ingress controller. The following example uses three nodes: `dockeree-worker-linux-1`, `dockeree-worker-linux-2`, and `dockeree-worker-linux-3`:
 
-    ![](../images/deploy-ingress-controller-1.png){: .with-border}
+```
+ 🐳  → kubectl get nodes
+NAME                           STATUS    ROLES     AGE       VERSION
+dockeree-worker-linux-1    Ready     <none>    5d        v1.8.11-docker-8d637ae
+dockeree-worker-linux-2    Ready     <none>    5d        v1.8.11-docker-8d637ae
+dockeree-worker-linux-3    Ready     <none>    5d        v1.8.11-docker-8d637ae
 
-## Create a grant
+```
 
-The default service account that's associated with the `ingress-nginx`
-namespace needs access to Kubernetes resources, so create a grant with
-`Restricted Control` permissions.
+The label `infra.role=ingress` is used to label the nodes:
```

Comment:
Please add that this is an arbitrary naming convention. Customers can choose anything they won't.

Reply:
Added some additional wording.